docroot/modules/custom/sitenow_advanced_webform/src/Plugin/WebformHandler/ProspectorRemotePostWebformHandler.php

```diff
@@ -0,0 +1,133 @@
+    $instance = parent::create($container, $configuration, $plugin_id, $plugin_definition);
+    $instance->httpClient = $container->get('http_client');
+    return $instance;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    parent::submitConfigurationForm($form, $form_state);
+    $this->applyFormStateToConfiguration($form_state);
+  }
+
+  public function prepareForm(
+    WebformSubmissionInterface $webform_submission,
+    $operation,
+    FormStateInterface $form_state
+  ) {
+    // @todo This doesn't seem quite right because it might show to an end user.
+    //   Maybe it should be limited by the user's permissions? Or maybe there is
+    //   a better place to show this message where only a form admin would see
+    //   it?
+    // Load configuration.
+    $config = $this->configFactory->get('sitenow_advanced_webform.settings');
+    $site_uuid = $config->get('prospector.site_uuid');
+    // We need a site UUID to proceed.
+    if (!$site_uuid) {
+      // Generate a URL for the sitenow_advanced_webform settings form.
+      $url = Url::fromRoute('sitenow_advanced_webform.settings_form')->toString();
+      // Print a message directing the user to the settings form.
+      $this->messenger()->addError($this->t('Prospector Site UUID is missing. Please visit the settings form at @url to configure.', [
+        '@url' => $url,
+      ]));
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function postSave(WebformSubmissionInterface $webform_submission, $update = TRUE) {
+    // Load configuration.
+    $config = $this->configFactory->get('sitenow_advanced_webform.settings');
+    $site_uuid = $config->get('prospector.site_uuid');
+    // We need a site UUID to proceed.
+    if (!$site_uuid) {
+      // @todo Log this?
+      return;
+    }
+
+    // Load basic auth credentials from config.
+    $auth = $config->get('prospector.auth');
+    // We need auth credentials to proceed.
+    if (!isset($auth['user']) || !isset($auth['pass'])) {
+      // @todo I think this needs to be moved similar to the site_uuid check.
+      $this->messenger()->addError($this->t('Prospector authentication credentials are missing. Please contact the SiteNow team for assistance.'));
+      return;
+    }
+    $options = [
+      'auth' => array_values($auth),
+    ];
+
+    // Add data from first_name, last_name, email, and phone fields.
+    $elements = ['first_name', 'last_name', 'email', 'phone'];
+    $data = [];
+    foreach ($elements as $element) {
+      if ($value = $webform_submission->getElementData($element)) {
+        $data[$element] = $value;
+      }
+    }
+    $data['siteInteractionUuid'] = $site_uuid;
+    $options['json'] = $data;
+
+    // Send http request.
+    try {
+      // @todo Should the URL be stored in config or as a setting?
```

Comment:
i feel like it should be a non-UI config for now similar to the sitenow_events endpoint as I imagine it will be changed soon and need to be updated across sites. ignored so we could do some site-specific testing if needed